Python/optimizer.c

```diff
@@ -378,13 +378,17 @@ translate_bytecode_to_trace(
     _Py_CODEUNIT *initial_instr = instr;
     int trace_length = 0;
     int max_length = buffer_size;
+    int reserved = 0;
```

Comment:
The point of reserve is to make sure that you won't error out in the middle of an opcode's translation to uops?

Reply:
Correct. I'd rather reserve space ahead than having to bail out in the middle and undo some work already done (the undoing feels more brittle).